include/wx/datetime.h

```diff
@@ -1728,9 +1728,11 @@ class WXDLLIMPEXP_BASE wxDateTimeChristianHolidays : public wxDateTimeUSCatholic
 
 inline bool wxDateTime::IsInStdRange() const
 {
-    // currently we don't know what is the real type of time_t so prefer to err
-    // on the safe side and limit it to 32 bit values which is safe everywhere
-    return m_time >= 0l && (m_time / TIME_T_FACTOR) < wxINT32_MAX;
+    // if sizeof(time_t) is greater than 32 bits, we assume it
+    // is safe to return values exceeding wxINT32_MAX
+
+    return m_time >= 0l &&
+        ( (sizeof(time_t) > 4 ) || ( (m_time / TIME_T_FACTOR) < wxINT32_MAX) );
```

Comment:
It looks like this ought to use use the same limit as the code in `datetime.cpp` (i.e. 2200 right now, but see there), shouldn't it?